torchvision/transforms/transforms.py

```diff
@@ -70,6 +71,140 @@ def __repr__(self):
         return format_string
 
 
+class MultiCompose(Compose):
+    """Like Compose, but processes tuples of images with the same random params.
+
+    That is, every image in a tuple/list will be transformed exactly the same way
+    (random rotation, crop, and so on).
+
+    Args:
+        transforms (list of ``Transform`` objects): list of transforms to compose.
+
+    Example:
+        >>> mc = transforms.MultiCompose([
+        >>>     transforms.CenterCrop(10),
+        >>>     transforms.Pad(1),
+        >>> ])
+        >>> imgs = mc([img1, img2, img3])
+    """
+    def __call__(self, imgs):
+        for t in self.transforms:
+            try:
+                params = t.generate_params(imgs[0])
+            except AttributeError:
+                params = None
+            # Non-random transforms will never generate any params, so we can call them
+            # in the old style "__call__(self, img)" instead of having to refactor them
+            # to take a "param" kwarg as well. This is only needed for actually random
+            # transforms.
+            if params is not None:
+                imgs = tuple(t(img, params) for img in imgs)
+            else:
+                imgs = tuple(t(img) for img in imgs)
+        return imgs
+
+
+class SegmentationCompose(MultiCompose):
+    """Composes multiple transforms together, specifically for segmentation data.
+
+    Assumes that the inputs are tuples of:
+        (image, label)
+    or even
+        (image, image, ..., image, label)
```

Comment:
While this is the most common case, it does not handle other (segmentation) use-cases that can appear, where we do not have a straight concept of `image` and `label`.
Imagine for example that we are predicting a depth map and a segmentation map from a pair of two images.
In this case, we would have two images and two labels, which is currently not supported by this approach.